File: i2f-jdk/i2f-swl/web/i2f-swl/src/i2f-turbo-web/i2f-swl/extension/impl/sm/antherd/SwlAntherdSm2AsymmetricEncryptor.js
import Sm2Util from "../../../../../i2f-core/crypto/asymmetric/Sm2Util";
import AsymKeyPair from "../../../../../i2f-core/crypto/asymmetric/AsymKeyPair";

/**
 * @param publicKey {String} base64
 * @param privateKey {String} base64
 * @return {SwlAntherdSm2AsymmetricEncryptor}
 * @implements {ISwlAsymmetricEncryptor}
 */
function SwlAntherdSm2AsymmetricEncryptor(publicKey = null, privateKey = null) {
    /**
     * @type {String} hex
     */
    this._publicKey = publicKey
    /**
     * @type {String} hex
     */
    this._privateKey = privateKey

}


/**
 * @return {AsymKeyPair}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.generateKeyPair = function () {
    return Sm2Util.genKeyPair()
}
/**
 * @return {AsymKeyPair}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.getKeyPair = function () {
    return new AsymKeyPair(this._publicKey, this._privateKey)
}
/**
 * @param key : {AsymKeyPair}
 * @return {void}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.setKeyPair = function (key) {
    this._publicKey = key.getPublicKey()
    this._privateKey = key.getPrivateKey()
}
/**
 * @return {String}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.getPublicKey = function () {
    return this._publicKey
}
/**
 * @param publicKey {String}
 * @return {void}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.setPublicKey = function (publicKey) {
    this._publicKey = publicKey
}
/**
 * @return {String}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.getPrivateKey = function () {
    return this._privateKey
}
/**
 *
 * @param privateKey {String}
 * @return {void}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.setPrivateKey = function (privateKey) {
    this._privateKey = privateKey
}
/**
 *
 * @param data {String}
 * @return {String}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.encrypt = function (data) {
    return Sm2Util.publicKeyEncrypt(this._publicKey, data)
}
/**
 *
 * @param data {String}
 * @return {String}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.decrypt = function (data) {
    return Sm2Util.privateKeyDecrypt(this._privateKey, data)
}
/**
 *
 * @param data {String}
 * @return {String}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.sign = function (data) {
    return Sm2Util.makeSign(this._privateKey, data)
}
/**
 *
 * @param sign {String}
 * @param data {String}
 * @return {boolean}
 */
SwlAntherdSm2AsymmetricEncryptor.prototype.verify = function (sign, data) {
    return Sm2Util.verifySign(this._publicKey, sign, data)
}

export default SwlAntherdSm2AsymmetricEncryptor

File: i2f-extension/i2f-extension-swl/src/main/java/i2f/extension/swl/impl/sm/antherd/SwlAntherdSm2AsymmetricEncryptor.java
package i2f.extension.swl.impl.sm.antherd;

import com.antherd.smcrypto.sm2.Keypair;
import i2f.crypto.std.encrypt.asymmetric.key.AsymKeyPair;
import i2f.extension.jce.sm.antherd.encrypt.asymmetric.Sm2Encryptor;
import i2f.swl.consts.SwlCode;
import i2f.swl.exception.SwlException;
import i2f.swl.std.ISwlAsymmetricEncryptor;

/**
 * @author Ice2Faith
 * @date 2024/7/10 9:45
 * @desc
 */
public class SwlAntherdSm2AsymmetricEncryptor implements ISwlAsymmetricEncryptor {
    private Sm2Encryptor encryptor = new Sm2Encryptor();

    @Override
    public AsymKeyPair generateKeyPair() {
        Keypair keypair = Sm2Encryptor.genKey();
        return new AsymKeyPair(keypair.getPublicKey(), keypair.getPrivateKey());
    }

    @Override
    public AsymKeyPair getKeyPair() {
        return encryptor.getAsymKeyPair();
    }

    @Override
    public void setKeyPair(AsymKeyPair key) {
        encryptor.setAsymKeyPair(key);
    }

    @Override
    public String getPublicKey() {
        return encryptor.getPublicKeyString();
    }

    @Override
    public void setPublicKey(String publicKey) {
        encryptor.setPublicKeyString(publicKey);
    }

    @Override
    public String getPrivateKey() {
        return encryptor.getPrivateKeyString();
    }

    @Override
    public void setPrivateKey(String privateKey) {
        encryptor.setPrivateKeyString(privateKey);
    }

    @Override
    public String encrypt(String data) {
        try {
            return encryptor.encrypt(data);
        } catch (Exception e) {
            throw new SwlException(SwlCode.ASYMMETRIC_ENCRYPT_EXCEPTION.code(), e.getMessage(), e);
        }
    }

    @Override
    public String decrypt(String data) {
        try {
            return encryptor.decrypt(data);
        } catch (Exception e) {
            throw new SwlException(SwlCode.ASYMMETRIC_DECRYPT_EXCEPTION.code(), e.getMessage(), e);
        }
    }

    @Override
    public String sign(String data) {
        try {
            return encryptor.sign(data);
        } catch (Exception e) {
            throw new SwlException(SwlCode.ASYMMETRIC_SIGN_EXCEPTION.code(), e.getMessage(), e);
        }
    }

    @Override
    public boolean verify(String sign, String data) {
        try {
            return encryptor.verify(sign, data);
        } catch (Exception e) {
            throw new SwlException(SwlCode.ASYMMETRIC_VERIFY_EXCEPTION.code(), e.getMessage(), e);
        }
    }
}
